spec/mongoid/query_cache_spec.rb

```diff
@@ -450,6 +450,19 @@
       Band.collection.indexes.create_one(name: 1)
     end
   end
+
+  context 'when the initial query does not exhaust the results' do
+    before do
+      Mongoid::QueryCache.enabled = true
```

Comment:
Usually it is good idea to temporarily enable config option and restore original state once test is done (for example using `around` block).

https://github.com/mongodb/mongoid/blob/b3a5918e504f4ac9ae4e750701bb02db78000960/spec/mongoid/query_cache_spec.rb#L308-L313

Reply:
Note that at the top of the description block, there is already an `around` block defined:

```ruby
around do |spec|
    Mongoid::QueryCache.clear_cache
    Mongoid::QueryCache.cache { spec.run }
  end
```

The implementation of `QueryCache.cache` already saves the value from before the passed block and restores it after.

Reply:
Is enabling query cache again on this line redundant then?